portfolio/src/main/webapp/script.js

```diff
@@ -71,3 +62,9 @@ function constructImage(imageIndex){
   txtContainer.innerHTML = '';
   txtContainer.appendChild(txtElement);
 } 
+
+async function fetchWordAsync() {
+  const response = await fetch('/data');
+  const word = await response.text();
+  document.getElementById('fetch-response').innerText = word;
```

Comment:
use innerHTML instead of innerText since the server is returning HTML you want to render.